### DISCUSSION TITLE
Table and related notes form together - is it possible?

### DISCUSSION THREAD

**Discussion by shirokovnv** [filamentphp/filament#7196] [Help] [answered]
Greetings.

Thanks for your effort and awesome filament package!

I have the following use case:

1. A table with some data, grouped by date (I have simple date filter).
2. Table rendered by the custom widget.
4. The model for table is read-only, it depends on MYSQL view.

I would like to add a "notes" form with one field - "body" below table. And "save" button.

The notes model also grouped by date (day), so, if I change date in table filter I would like to see changes in my form field.

If I press the save button, I would like to save notes data to the DB.

Can I achieve this with filament ? 

Or it can be done way simpler, than I described?

PHP version: `8.1`
Filament version: `2.17.50`
Laravel Version: `9.52.10`

Thanks.

**Comment by pboivin** [comment]
I think this can be done, but I'm not sure I get the full picture... can you share a screenshot or some mockup if you have it?
From what I understand, you'll need a few things:
- Add a [custom page](https://filamentphp.com/docs/2.x/admin/pages/getting-started) to wrap everything
- In the header, add a [table widget](https://filamentphp.com/docs/2.x/admin/dashboard/tables)
- Below the header, in the main page area, add a Livewire component that contains both a [custom table](https://filamentphp.com/docs/2.x/tables/getting-started#preparing-your-livewire-component) and a [custom form](https://filamentphp.com/docs/2.x/forms/getting-started#preparing-your-livewire-component)

Did I get this right, you need 2 tables in the same page?

**Reply by shirokovnv** [reply, marked as answer]
Yes, that's correct.

To be honest, I already found a way to do it.

I used custom table widget, a filter and an action, so they work together like this:

1. The action represents form with notes.
2. When the date changes in filter, the action knows about it (from mount state callback) and change self-state to load needed one.
3. Save action button saves the notes.

Code fragment:

```php
/**
     * @return array
     *
     * @throws \Exception
     */
    protected function getTableFilters(): array
    {
        return [
            Filter::make(self::KEY_FILTER_NAME)
                ->form([
                    Forms\Components\DatePicker::make(self::KEY_FILTER_PARAM)
                        ->default(now()),
                ])
                ->default(),
        ];
    }

    /**
     * @return array
     * 
     * @throws \Exception
     */
    protected function getTableHeaderActions(): array
    {
        return [
            // Day notes
            Tables\Actions\Action::make('Day notes')
                ->action(function (array $data, DayNoteService $service): void {
                    $service->updateBody($data['day_note_id'], $data['day_note_body']);
                })
                ->mountUsing(function (Forms\ComponentContainer $form, DayNoteService $service) {
                    $selected_date = $this->tableFilters['date_filter']['date'] ?? Carbon::now()->format('Y-m-d');
                    $day_note = $service->findOrCreateByDate($selected_date);

                    $form->fill([
                        'day_note_id' => $day_note->id,
                        'day_note_body' => $day_note->body,
                    ]);
                })
                ->form([
                    Forms\Components\Hidden::make('day_note_id'),
                    Forms\Components\Textarea::make('day_note_body')
                        ->label('Content'),
                ]),
        ];
    }
```

The form not below the table (by condition above), but that's not a big deal.

Thank you for your answer anyway!! 

And please excuse me, I should update this post earlier, not wasting your time.

I hope this helps the community, almost a little.
